lms/djangoapps/oauth_dispatch/views.py

```diff
@@ -36,7 +36,11 @@ def get_adapter(self, request):
         """
         Returns the appropriate adapter based on the OAuth client linked to the request.
         """
-        if dot_models.Application.objects.filter(client_id=self._get_client_id(request)).exists():
+        dot_application = dot_models.Application.objects.filter(client_id=self._get_client_id(request))
+        if dot_application.exists():
+            if isinstance(self, RevokeTokenView):
+                request.POST = request.POST.copy()  # Make a mutable copy of the POST parameters.
+                request.POST['client_secret'] = dot_application[0].client_secret
```

Comment:
@wajeeha-khalid This doesn't seem right to me.  We are bypassing security measures for Public Clients by implicitly providing them a client_secret when they didn't provide it.  Let me take a look at the DOT code.